pilot/pkg/model/telemetry.go

```diff
@@ -542,6 +545,13 @@ func mergeMetrics(metrics []*tpb.Metrics, mesh *meshconfig.MeshConfig) map[strin
 			// For each override, we normalize the configuration. The metrics list is an ordered list - latter
 			// elements have precedence. As a result, we will apply updates on top of previous entries.
 			for _, o := range m.Overrides {
+				// // if we disable all metrics, we should drop the entire filter
+				if isAllMetrics(o.GetMatch()) && o.Disabled.GetValue() {
```

Comment:
what if root namesapce disables all, but then a later config enables them?

Reply:
it would depends on how `applicableTelemetries`  works, let me check that.

Reply:
updated